### ISSUE TITLE
[events] Consider events enumeration

### ISSUE THREAD

**Issue by kouks** [exteranto/framework#6]
Would be nice to standardise all native/framework events under an enumeration instead of using the `app.package.event` notation.

**Comment by bausano** [comment]
I was thinking about this and I suppose moving onwards when having lots of modules that all emit events this is going to be necessary for both IDE hinting of native events, avoiding event overlapping (maybe providers would be responsible for event registering?) and consistency among packages (this is more of our internal issue, not entirely on experanto, but yeah).

Might do for an interesting read:
https://github.com/Microsoft/TypeScript/issues/17592

Maybe something like `AppEvent.MANAGEMENT.RUNTIME.INSTALLED` where `MANAGEMENT` would be static get of class `AppEvent` that extends some `EventEnum` empty interface.

Then in each non-core module you'd register your own implementation:
```
class PromotionEvent implements EventEnum {
  public static get CLOSED () : string {
    return 'app.promotions.closed'
  }
}

register () : void {
  ...
  // Wouldn't put it on dispatcher maybe, just throwing my 5 cents in.
  dispatcher.registerEventTree(new PromotionEvent())
}
```

and in event registration you could use is like this
```
import { PromotionEvent } from '...'

// Doing stuff.

dispatcher.fire(PromotionEvent.OPEN)
```

and `registerEventTree` would then call all those static methods recursively until it gets all strings such as 'app.promotions.closed', checked that this event is not already in the set of registered events (from another module, ...) and added it there.

This would I guess force developers to think about their event structure and also allowed anyone just to open the file with `PromotionEvent` to see all events.

This is just a random thing that now came to my mind, it e.g. would have to be changed somewhat to get IDE hinting. But I think it's a step in a right direction and maybe you can think of something better based on this.

**Comment by bausano** [comment]
closed by #43
